packages/amplify-provider-awscloudformation/lib/transform-graphql-schema.js

```diff
@@ -41,35 +41,45 @@ async function transformerVersionCheck(
   updatedResources, usedDirectives,
 ) {
   const versionChangeMessage = 'The default behaviour for @auth has changed in the latest version of Amplify\nRead here for details: https://aws-amplify.github.io/docs/cli-toolchain/graphql#authorizing-subscriptions';
-  let transformerConfig;
+  const checkVersionExist = config => (config && config.Version);
+
   // this is where we check if there is a prev version of the transformer being used
   // by using the transformer.conf.json file
-  try {
-    transformerConfig = await readTransformerConfiguration(cloudBackendDirectory);
-  } catch (err) {
-    // check local resource if the question has been answered before
-    transformerConfig = await readTransformerConfiguration(resourceDir);
-  }
+  const cloudTransformerConfig = await readTransformerConfiguration(cloudBackendDirectory);
+  const cloudVersionExist = checkVersionExist(cloudTransformerConfig);
+
+  // check local resource if the question has been answered before
+  const localTransformerConfig = await readTransformerConfiguration(resourceDir);
```

Comment:
this would fail when adding api for the first time - if we want to catch this at the level of transformConfig I've added a pr [2354](https://github.com/aws-amplify/amplify-cli/pull/2354) to address that